generators/client/templates/react/src/main/webapp/app/modules/account/register/register.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react';
+import { Translate, translate } from 'react-jhipster';
+import { connect } from 'react-redux';
+
+import { AvForm, AvField } from 'availity-reactstrap-validation';
+import { Row, Col, Alert, Button } from 'reactstrap';
+
+import { register, reset } from '../../../reducers/register';
+
+export const mainErrorMessages = {
+  failed: (
+    <Translate contentKey="register.messages.error.fail">
+      <strong>Registration failed!</strong> Please try again later.
+    </Translate>
+  ),
+  userexists: (
+    <Translate contentKey="register.messages.error.userexists">
+      <strong>Login name already registered!</strong> Please choose another one.
+    </Translate>
+  ),
+  emailexists: (
+    <Translate contentKey="register.messages.error.emailexists">
+      <strong>Email is already in use!</strong> Please choose another one.
+    </Translate>
+  )
+};
+
+export interface IRegisterProps {
+  register: Function;
+  reset: Function;
+  registrationSuccess: boolean;
+  registrationFailure: boolean;
+  errorMessage: string;
+}
+
+export class RegisterPage extends React.Component<IRegisterProps, {}> {
+  constructor(props) {
+    super(props);
+    this.handleValidSubmit = this.handleValidSubmit.bind(this);
+  }
+
+  componentWillUnmount() {
+    this.props.reset();
+  }
+
+  handleValidSubmit = (event, values) => {
+    this.props.register(values.username, values.email, values.firstPassword);
+    event.preventDefault();
+  };
+
+  render() {
+    const { registrationSuccess, registrationFailure, errorMessage } = this.props;
+    let alertMessage = null;
+
+    if (registrationFailure) {
+      alertMessage = (
+        <Alert color="danger">
+          {mainErrorMessages[errorMessage] === undefined ? mainErrorMessages.failed : mainErrorMessages[errorMessage]}
```

Comment:
just do a truthy check as it covers undefined as well `{mainErrorMessages[errorMessage] ? mainErrorMessages.failed : mainErrorMessages[errorMessage]}`